src/lakefs_spec/transaction.py

```diff
@@ -25,30 +26,43 @@
 
 
 @dataclass
-class Placeholder(Generic[T]):
+class Placeholder(wrapt.ObjectProxy, Generic[T]):
     """A generic placeholder for a value computed by the lakeFS server in a versioning operation during a transaction."""
 
-    value: T | None = None
-    """The abstract value. Set only on completion of the versioning operation during the defining transaction."""
+    _expect: Type[T] | None = None
 
-    def available(self):
+    def __init__(self, wrapped: T | None = None, expect: Type[T] | None = None):
+        super(Placeholder, self).__init__(wrapped)
+        self._expect = expect
+
+    @property
+    def available(self) -> bool:
         """Whether the wrapped value is available, i.e. already computed."""
-        return self.value is not None
+        return self.__wrapped__ is not None
 
     def set_value(self, value: T) -> None:
         """Fill in the placeholder. Not meant to be called directly except in the completion of the transaction."""
-        self.value = value
+        self.__wrapped__ = value
 
-    def unwrap(self) -> T:
-        """Return the placeholder's value after it has been filled."""
-        if self.value is None:
-            raise RuntimeError("placeholder unfilled")
-        return self.value
+    def __getattr__(self, name):
+        """Override __getattr__ to access the attributes of the wrapped object directly."""
+        if self.__wrapped__ is None:
+            raise RuntimeError(
+                f" '{name}' attribute of object '{type(self).__name__}' is unfilled."
+            )
+        return getattr(self.__wrapped__, name)
```

Comment:
I'd prefer all of these to be class method calls instead of builtins, i.e. `self.__wrapped__.__getattr__(name)`, and similarly below with `__repr__` and `__str__`.

Reply:
Sure. 

Why your preference of methods over builtins?

Reply:
Adressed in [cebc153](https://github.com/aai-institute/lakefs-spec/pull/225/commits/cebc153db0f3087e5a87d2d9382f1c1b4dcd80b8)